pkg/broker/provision_job.go

```diff
@@ -51,18 +52,19 @@ func (m ProvisionMsg) Render() string {
 }
 
 // NewProvisionJob - Create a new provision job.
-func NewProvisionJob(serviceInstance *apb.ServiceInstance, clusterConfig apb.ClusterConfig,
+func NewProvisionJob(task string, serviceInstance *apb.ServiceInstance, clusterConfig apb.ClusterConfig,
 	log *logging.Logger,
 ) *ProvisionJob {
 	return &ProvisionJob{
 		serviceInstance: serviceInstance,
 		clusterConfig:   clusterConfig,
-		log:             log}
+		log:             log,
+		task:            task}
 }
 
 // Run - run the provision job.
 func (p *ProvisionJob) Run(token string, msgBuffer chan<- WorkMsg) {
-	podName, extCreds, err := apb.Provision(p.serviceInstance, p.clusterConfig, p.log)
+	podName, extCreds, err := apb.Provision(p.task, p.serviceInstance, p.clusterConfig, p.log)
```

Comment:
I'm not a huge fan of parameterizing `Provision` with a couple variants of behavior; it breaks the semantics of the function from a caller's perspective. One thought would be to have two distinct `Provision` and `Update` available on the public interface, and behind the scenes they actually do some specific behavior for their case, and call into some shared function.